api/dataset.go

```diff
@@ -94,8 +94,56 @@ func (api *API) getDatasetJSON(ctx context.Context, r *http.Request, params *dat
 	return response, nil
 }
 
+func (api *API) validateGeography(ctx context.Context, r *http.Request, params *datasetParams, datasetRequest cantabular.StaticDatasetQueryRequest) ([]string, *[]cantabular.Filter, error) {
+	geographyQuery := strings.Split(r.URL.Query().Get("area-type"), ",")
+
+	if len(geographyQuery) < 1 {
+		return nil, nil, errors.New("unable to locate area-type")
+	}
+
+	geography := geographyQuery[0]
+	geographyOptions := geographyQuery[1:]
+
+	foundGeography := false
+	for _, d := range params.geoDimensions {
+		if strings.EqualFold(d, geography) {
+			foundGeography = true
+			params.sortedDimensions[0] = geographyQuery[0]
+
+			datasetRequest.Variables = params.sortedDimensions
+			break
+		}
+	}
+
+	if len(geographyOptions) > 0 {
+		// Check the area code is in the area type
+		for _, a := range geographyOptions {
+			cReq := cantabular.GetAreaRequest{
+				Dataset:  params.basedOn,
+				Variable: geography,
+				Category: a,
+			}
+
+			_, err := api.ctblr.GetArea(ctx, cReq)
```

Comment:
Isn't there a `cantabular.GetAreas` function that could query all at once?

Reply:
This is a test to validate that all geography options are valid. There is a possibility of getting all the areas i.e. GetAreas, and then check if geography options are valid in  list returned by GetAreas, but GetAreas is based on pagination and will have to implement pagination logic to retrieve all the list. 
So unless geography options is a big list and cantabular.GetArea is inefficient, there  is no harm in the way it is currently implemented.